static/modules/entity.js

```diff
@@ -114,18 +128,76 @@ export class Entity {
     this.vel = vel;
     /** @type {Vector} */
     this.acc = acc;
+    this.collisionType = "Circle";
+  }
+
+  /**
+   * Set the collisionShape of the entity. Give no arguments to reset the shape
+   * to be calculated each call of getCollisonShape
+   * @param {CollisionShape | undefined} collisionShape
```

Comment:
This typing `{CollisionShape | undefined}` is redundant I think.
You should be able to just do `@param {CollisionShape} [collisionShape]` meaning it's optional (can be undefined)